include/deal.II/lac/utilities.h

```diff
@@ -0,0 +1,301 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__lac_utilities_h
+#define dealii__lac_utilities_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/lac/lapack_templates.h>
+#include <deal.II/lac/lapack_support.h>
+#include <deal.II/lac/vector_memory.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Utilities
+{
+  /**
+   * A collection of linear-algebra utilities.
+   */
+  namespace LinearAlgebra
+  {
+
+    /**
+     * Estimate an upper bound for the largest eigenvalue of @p op by a @p k -step
+     * Lanczos process starting from the initial vector @p v0. Typical
+     * values of @p k are below 10. The returned
+     * value is $ ||T||_2 + ||f||_2$, where $H V_k=V_k T_k+f_k e_k^T$ and
```

Comment:
I suspect that `T` in the norm is the final value of the matrix `T_k`? But what are H, V_k, f_k and e_k in this formula?